plugin/clone/src/clone_common.cc

```diff
@@ -0,0 +1,106 @@
+/*
+  Copyright (C) 2022, Laurynas Biveinis
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; version 2 of the License.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA */
+
+#include "plugin/clone/include/clone_common.h"
+
+#include <string>
+#include <utility>
+
+#include "lex_string.h"
+#include "my_dbug.h"
+#include "my_inttypes.h"
+#include "plugin/clone/include/clone_hton.h"
+#include "sql/handler.h"
+#include "sql/json_dom.h"
+#include "sql/sql_plugin_ref.h"
+#include "storage/perfschema/table_log_status.h"
+
+namespace myclone {
+
+int Ha_clone_common_cbk::precopy_and_synchronize_engines(THD *thd,
+                                                         uint task_id) {
+  const auto &all_locators = get_all_locators();
+
+  // Must be called by InnoDB the clone-driving engine
+  assert(all_locators[0].m_hton->db_type == DB_TYPE_INNODB);
```

Comment:
What happens if we move to a MyRocks only scenario without InnoDB? How much of the InnoDB Clone Task manager code is needed for other engines to be independent of InnoDB?

Reply:
@hermanlee nothing from the InnoDB clone task manager will be required in an InnoDB-less instance. Code like [code_hton.cc:make_innodb_first](https://github.com/facebook/mysql-5.6/pull/1250/files#diff-085fb71931c000997079f16e6b8459530c72c418a95a88e848f17c315e686b20R108-R128) would not be required neither. (I expect it will be kept but called conditionally depending on what SEs are registered)

Reply:
Once the InnoDB clone task manager disappears, those stages will disappear too, as it is defined by them, not by the plugin. It will be up to MyRocks to define some stages from scratch., i.e. SST_COPY and FINAL_COPY , and add the required code for stats.